static/js/vue-cdr-access/tests/unit/facetTags.spec.js

```diff
@@ -0,0 +1,156 @@
+        await wrapper.vm.$nextTick();
+        const selected_tags = wrapper.findAll('.search-text');
+        expect(selected_tags.at(0).text()).toMatch(/testCollection/);
+        expect(selected_tags.at(1).text()).toMatch(/test2Collection/);
+    });
+
+    it("displays multiple selected tags of different types", async () => {
+        wrapper.vm.$router.push('/search/?collection=d77fd8c9-744b-42ab-8e20-5ad9bdf8194e||' +
+        '88386d31-6931-467d-add5-1d109f335302&format=image||text');
+        await wrapper.vm.$nextTick();
+        const selected_tags = wrapper.findAll('.search-text');
+        expect(selected_tags.at(0).text()).toMatch(/testCollection/);
+        expect(selected_tags.at(1).text()).toMatch(/test2Collection/);
+        expect(selected_tags.at(2).text()).toMatch(/image/);
+        expect(selected_tags.at(3).text()).toMatch(/text/);
+    });
+
+    it("clears a selected tag", async () => {
+        wrapper.vm.$router.push('/search/?collection=d77fd8c9-744b-42ab-8e20-5ad9bdf8194e');
+        await wrapper.vm.$nextTick();
+        expect(wrapper.find('.search-text').text()).toMatch(/testCollection/);
+
+        wrapper.find('.search-text').trigger('click');
+        await wrapper.vm.$nextTick();
+        expect(wrapper.find('.search-text').exists()).toBe(false);
+    });
+
+    it("clears sub tags of a selected tag", async () => {
+        wrapper.vm.$router.push('/search/?format=image||image/png');
+        await wrapper.vm.$nextTick();
+        const selected_tags = wrapper.findAll('.search-text');
+        expect(selected_tags.at(0).text()).toMatch(/image/);
+        expect(selected_tags.at(1).text()).toMatch(/image\/png/);
+
+        wrapper.find('.search-text').trigger('click'); // image tag
+        await wrapper.vm.$nextTick();
+        expect(wrapper.find('.search-text').exists()).toBe(false);
+    });
+
+    it("does not clear parent tag of a selected tag", async () => {
+        wrapper.vm.$router.push('/search/?format=image||image/png');
+        await wrapper.vm.$nextTick();
+        let selected_tags = wrapper.findAll('.search-text');
+        expect(selected_tags.at(0).text()).toMatch(/image/);
+        expect(selected_tags.at(1).text()).toMatch(/image\/png/);
+
+        selected_tags.at(1).trigger('click'); // image/png tag
+        await wrapper.vm.$nextTick();
+        selected_tags = wrapper.findAll('.search-text');
+        expect(selected_tags.at(0).text()).toMatch(/image/);
+        expect(selected_tags.length).toEqual(1);
+    });
+
+    it("displays search query", async () => {
+        wrapper.vm.$router.push('/search/?anywhere=stones');
+        await wrapper.vm.$nextTick();
+        expect(wrapper.find('.search-text').text()).toMatch(/stone/);
+    });
+
+    it("does not display search query if empty", async () => {
+        wrapper.vm.$router.push('/search/?anywhere=');
+        await wrapper.vm.$nextTick();
+        expect(wrapper.find('.search-text').exists()).toBe(false);
+    });
+});
```

Comment:
Add tests for one of the date term fields to make sure the formatting is working.

Reply:
Also add test for searching the title index, and searching both the title and the keyword index at the same time.